packages/code-editor/src/index.js

```diff
@@ -0,0 +1,335 @@
+/**
+ * External dependencies
+ */
+import { isObject, filter, isEqual, without, contains, delay } from 'lodash';
+import jQuery from 'jquery';
+import CodeMirror from 'codemirror';
+
+/**
+ * Default settings for code editor.
+ *
+ * @since 4.9.0
+ * @type {object}
+ */
+export const defaultSettings = {
+	codemirror: {},
+	csslint: {},
+	htmlhint: {},
+	jshint: {},
+	onTabNext() {},
+	onTabPrevious() {},
+	onChangeLintingErrors() {},
+	onUpdateErrorNotice() {},
+};
+
+/**
+ * Configure linting.
+ *
+ * @param {CodeMirror} editor - Editor.
+ * @param {Object}     settings - Code editor settings.
+ * @param {Object}     settings.codeMirror - Settings for CodeMirror.
+ * @param {Function}   settings.onChangeLintingErrors - Callback for when there are changes to linting errors.
+ * @param {Function}   settings.onUpdateErrorNotice - Callback to update error notice.
+ *
+ * @return {void}
+ */
+function configureLinting( editor, settings ) {
+	let currentErrorAnnotations = [],
+		previouslyShownErrorAnnotations = [];
+
+		/**
```

Comment:
Tabbing alignment (various related issues with most functions in this file).

Reply:
You know, old style WP scripts 🤷‍♂️ I did some tweaks, but there more we can do here :)